urllib3/exceptions.py

```diff
@@ -196,6 +196,14 @@ class ResponseNotChunked(ProtocolError, ValueError):
     pass
 
 
+class BodyNotHttplibCompatible(AssertionError, ValueError):
```

Comment:
Hmmm - not sure on ValueError. It seems like we probably want something similar to the existing RequestNotChunked exception.

But we should certainly inheriting from some form of urllib3's `HTTPError`.